lib/pbench/test/functional/server/test_user.py

```diff
@@ -15,50 +20,71 @@ class TestUser:
     had a known Administrative user (e.g., admin/admin).
     """
 
-    USERNAME = "functional"
-    PASSWORD = "tests"
-
-    def test_login_fail(self, pbench: Pbench):
+    def test_login_fail(self, pbench_server_client: PbenchServerClient):
         """
         Trying to log in to a non-existent user should fail.
 
         NOTE: This will fail if "nouser" exists.
         """
-        with pytest.raises(HTTPError) as e:
-            pbench.login("nouser", "nopassword")
-        assert (
-            str(e.value)
-            == "401 Client Error: UNAUTHORIZED for url: http://10.1.170.201/api/v1/login"
-        )
-
-    def test_register(self, pbench: Pbench):
+        with pytest.raises(
+            HTTPError, match=r"UNAUTHORIZED for url: http://.*?/api/v1/login"
```

Comment:
Why? The server doesn't support `https`. Right now, that would be a bug. Eventually it'll be required, and we'll complain about `http`. Let's worry about that then since at this point the test has no way to distinguish right from wrong.

Reply:
From a testing perspective, I think all that we care about here is that the status (e.g., `UNAUTHORIZED`) and the URI (e.g., `api/v1/login`) are correct; I don't think we care much about the rest of it.  However, the code in `connect()` goes to some trouble to preserve the access scheme if it is specified, and the rest of the code in this PR doesn't seem to care whether it is `http` or `https`, so it seemed to me that the test code should be similarly agnostic, which is easy to attain simply by adding a `s?` in the right spot in the match string.

Reply:
Or, we could just remove the `http://` and let the `.*` match it.